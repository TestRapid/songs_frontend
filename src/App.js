import { lazy, Suspense } from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import { CircularProgress, Typography } from "@material-ui/core";

// importing redux store
import Store from "./Store";

// custom components
import PrivateRoute from "./Components/PrivateRoute";
import Layout from "./Components/Layout";

// importing pages
import Home from "./Pages/Home/Home";
import Login from "./Pages/Auth/Login";
import Signup from "./Pages/Auth/Signup";
import Profile from "./Pages/Profile/Profile";

// importing css
import "./App.css";

// lazy loaded components;
const AllSongs = lazy(() => import("./Pages/AllSongs/AllSongs"));
const About = lazy(() => import("./Pages/About/About"));
const AddSong = lazy(() => import("./Pages/AddSong/AddSong"));
const Song = lazy(() => import("./Pages/Song/Song"));
const MySongs = lazy(() => import("./Pages/Mysongs/MySongs"));
const Playlist = lazy(() => import("./Pages/Playlist/Playlist"));

export function App() {
	return (
		<Store>
			<Router>
				<Layout>
					<Switch>
						<Route exact path="/">
							<Home />
						</Route>
						<Route path="/login">
							<Login />
						</Route>
						<Route path="/signup">
							<Signup />
						</Route>
						<Route path={`/songs/:id`}>
							<Suspense fallback={<CircularProgress />}>
								<Song />
							</Suspense>
						</Route>
						<Route path="/songs">
							<Suspense fallback={<CircularProgress />}>
								<AllSongs />
							</Suspense>
						</Route>
						<Route path="/about">
							<Suspense fallback={<CircularProgress />}>
								<About />
							</Suspense>
						</Route>
						<PrivateRoute path="/profile">
							<Profile />
						</PrivateRoute>
						<PrivateRoute path="/editsong/:id">
							<Suspense fallback={<CircularProgress />}>
								<AddSong />
							</Suspense>
						</PrivateRoute>
						<PrivateRoute path="/addsong">
							<Suspense fallback={<CircularProgress />}>
								<AddSong />
							</Suspense>
						</PrivateRoute>
						<PrivateRoute path="/mysongs">
							<Suspense fallback={<CircularProgress />}>
								<MySongs />
							</Suspense>
						</PrivateRoute>
						<PrivateRoute path="/playlist/:id">
							<Suspense fallback={<CircularProgress />}>
								<Playlist />
							</Suspense>
						</PrivateRoute>
						<Route paht="*">
							<Typography variant="h5" component="h5">
								Page not found
							</Typography>
						</Route>
					</Switch>
				</Layout>
			</Router>
		</Store>
	);
}

export default App;
