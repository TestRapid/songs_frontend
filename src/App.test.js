import { mount } from "enzyme";
import App from "./App";

describe("Test suite for snapshot test full dom", () => {
	let wrapper;

	beforeEach(() => {
		wrapper = mount(<App />);
	});

	it("test for snapshot", () => {
		expect(wrapper).toMatchSnapshot();
	});
});
