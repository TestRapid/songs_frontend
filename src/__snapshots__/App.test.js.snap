// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for snapshot test full dom test for snapshot 1`] = `
<App>
  <Store>
    <Provider
      store={
        Object {
          "@@observable": [Function],
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <BrowserRouter>
        <Router
          history={
            Object {
              "action": "POP",
              "block": [Function],
              "createHref": [Function],
              "go": [Function],
              "goBack": [Function],
              "goForward": [Function],
              "length": 1,
              "listen": [Function],
              "location": Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              },
              "push": [Function],
              "replace": [Function],
            }
          }
        >
          <Layout>
            <div
              className="makeStyles-loader-2"
            >
              <WithStyles(ForwardRef(CircularProgress))
                size="10rem"
              >
                <ForwardRef(CircularProgress)
                  classes={
                    Object {
                      "circle": "MuiCircularProgress-circle",
                      "circleDeterminate": "MuiCircularProgress-circleDeterminate",
                      "circleDisableShrink": "MuiCircularProgress-circleDisableShrink",
                      "circleIndeterminate": "MuiCircularProgress-circleIndeterminate",
                      "circleStatic": "MuiCircularProgress-circleStatic",
                      "colorPrimary": "MuiCircularProgress-colorPrimary",
                      "colorSecondary": "MuiCircularProgress-colorSecondary",
                      "determinate": "MuiCircularProgress-determinate",
                      "indeterminate": "MuiCircularProgress-indeterminate",
                      "root": "MuiCircularProgress-root",
                      "static": "MuiCircularProgress-static",
                      "svg": "MuiCircularProgress-svg",
                    }
                  }
                  size="10rem"
                >
                  <div
                    className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
                    role="progressbar"
                    style={
                      Object {
                        "height": "10rem",
                        "width": "10rem",
                      }
                    }
                  >
                    <svg
                      className="MuiCircularProgress-svg"
                      viewBox="22 22 44 44"
                    >
                      <circle
                        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                        cx={44}
                        cy={44}
                        fill="none"
                        r={20.2}
                        strokeWidth={3.6}
                        style={Object {}}
                      />
                    </svg>
                  </div>
                </ForwardRef(CircularProgress)>
              </WithStyles(ForwardRef(CircularProgress))>
            </div>
          </Layout>
        </Router>
      </BrowserRouter>
    </Provider>
  </Store>
</App>
`;
